modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -1490,4 +1499,42 @@ protected String getEstimatedExtentFunction() {
                 ? "ST_EstimatedExtent"
                 : "ST_Estimated_Extent";
     }
+
+    public Filter[] splitFilter(Filter filter, SimpleFeatureType schema) {
+        PostPreProcessFilterSplittingVisitor splitter =
+                new PostPreProcessFilterSplittingVisitor(
+                        dataStore.getFilterCapabilities(), schema, null) {
+                    private FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+
+                    @Override
+                    public Object visit(Function expression, Object notUsed) {
+                        if (expression instanceof JsonPointerFunction) {
+                            // takes the json pointer param to check if
+                            // can be encoded
+                            Expression param = expression.getParameters().get(1);
+                            if ((expression.getParameters().get(1) instanceof Literal))
+                                // can encode
+                                fcs.addType(JsonPointerFunction.class);
+                            else {
+                                FilterAttributeExtractor extractor = new FilterAttributeExtractor();
```

Comment:
The filter to sql will fail if the expression is not a Literal though (class cast exception).

Reply:
Never mind, I see you are replacing it with a Literal... however, you should not be modifying the filter here, unless you are sure it has been through a copy first, otherwise you risk changing some cached state (e.g., the structure of a in memory SLD representation, or some such).

Reply:
Thanks for your suggestion @aaime fixed